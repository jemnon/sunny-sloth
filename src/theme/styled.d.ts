import 'styled-components';
import {
  Breakpoints,
  Colors,
  ColorMode,
  FocusColors,
  Fonts,
  FontSizes,
  Header,
  HoverColors,
  MaxWidths,
  Spacing,
  ZIndex,
} from '../types/theme';

declare module 'styled-components' {
  export interface DefaultTheme {
    breakpoints: Breakpoints;
    colors: Colors;
    focusColors: FocusColors;
    fonts: Fonts;
    fontSizes: FontSizes;
    header: Header;
    hoverColors: HoverColors;
    maxWidths: MaxWidths;
    spacing: Spacing;
    transition: string;
    zIndex: ZIndex;
  }
}
