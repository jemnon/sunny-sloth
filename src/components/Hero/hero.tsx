import React, { FC } from 'react';
import styled from 'styled-components';
import Img from 'gatsby-image';
import { Images } from '../../types/images';
import logo from '../../images/svg/logo-white-horizontal.svg';

interface HeroProps {
  images?: Images[];
}

const HeroRoot = styled.div`
  position: relative;
  background-color: #000;
  height: 100vh;
  > div {
    opacity: 0.4;
    object-fit: cover;
    height: 100%;
  }
`;

const HeroLogo = styled.img`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  z-index: ${({ theme }): string => theme.zIndex.z1};
`;

const Hero: FC<HeroProps> = ({ images }) => {
  const [{ fluid }] = images || [];
  return (
    <HeroRoot id="hero">
      <HeroLogo src={logo} />
      <Img fluid={fluid} />
    </HeroRoot>
  );
};

export default Hero;
