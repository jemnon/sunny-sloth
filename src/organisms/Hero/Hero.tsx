import React, { forwardRef, ReactNode, useRef, useState } from 'react';
import styled from 'styled-components';
import DownButton from '../../atoms/DownButton';
import HeroContent from '../../molecules/HeroContent';
import useWindowResize from '../../hooks/useWindowResize';

interface HeroProps {
  title?: string;
  onViewPost?: () => void;
  image?: ReactNode;
}

const HeroContainer = styled.section`
  position: relative;
  background-color: ${({ theme }): string => theme.colors.nearBlack};
  overflow: hidden;
  height: 100vh;
`;

const HeroImgWrapper = styled.div`
  position: relative;
  opacity: 0.4;
  height: 100%;
  z-index: ${({ theme }): string => theme.zIndex.z0};
  > * {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
`;

const Hero = forwardRef<HTMLDivElement, HeroProps>(
  ({ image, title, onViewPost }, heroRef) => {
    const initialHeight =
      typeof window !== 'undefined' ? window.innerHeight : 0;
    const [height, setHeight] = useState<number | string>(initialHeight);
    const handleResize = (
      width?: number | string,
      height?: number | string,
    ): void => {
      if (height) setHeight(height);
    };
    const handleDownScroll = (): void => {
      if (height && typeof height === 'number') {
        window.scrollBy({ top: height - 64, left: 0, behavior: 'smooth' });
      }
    };
    useWindowResize(handleResize);
    return (
      <HeroContainer ref={heroRef}>
        {title && <HeroContent isCentered title={title} onClick={onViewPost} />}
        {image && <HeroImgWrapper>{image}</HeroImgWrapper>}
        <DownButton isCentered onClick={handleDownScroll} />
      </HeroContainer>
    );
  },
);

export default Hero;
