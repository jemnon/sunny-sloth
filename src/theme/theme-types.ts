export interface Breakpoints {
  sm: string;
  md: string;
  lg: string;
  xlg: string;
}

export interface Fonts {
  lato: string;
  latoBold: string;
  latoItalic: string;
  noto: string;
  notoBold: string;
}

export interface Colors {
  cream: string;
  darkTeal: string;
  teal: string;
  lightBrown: string;
  nearBlack: string;
  nearWhite: string;
  darkOrange: string;
  orange: string;
  white: string;
  red: string;
  green: string;
}

export type ColorMode = 'dark' | 'light';

export interface MaxWidths {
  sm: string;
  md: string;
  lg: string;
}

export interface Spacing {
  sm1: string;
  sm2: string;
  sm3: string;
  sm4: string;
  md1: string;
  md2: string;
  md3: string;
  md4: string;
  lg1: string;
  lg2: string;
  lg3: string;
  lg4: string;
}

export interface ZIndex {
  z0: string;
  z1: string;
  z2: string;
  z3: string;
  z4: string;
  z5: string;
  z6: string;
  z999: string;
  z9999: string;
}
