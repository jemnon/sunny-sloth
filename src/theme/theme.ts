const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
