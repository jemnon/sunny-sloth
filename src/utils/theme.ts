import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
