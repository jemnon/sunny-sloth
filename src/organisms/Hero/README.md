# Hero

### Description
Component description

```js
import Hero from '@fenderdigital/ui/FILL IN TYPEHero';
```

### Props
* `propName` - property name and description 
