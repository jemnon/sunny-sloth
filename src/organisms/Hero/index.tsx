import Hero from './Hero-ass';

export default Hero;
