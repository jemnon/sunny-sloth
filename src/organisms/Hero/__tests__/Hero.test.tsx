import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Hero from '../Hero';
import theme from '../../../theme';

describe('<Hero />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Hero />
    </ThemeProvider>,
  );
  it('exists', (): void => {
    expect(Hero).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
