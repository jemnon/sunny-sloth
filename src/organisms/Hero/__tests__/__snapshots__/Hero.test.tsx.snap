// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> renders defaults correctly 1`] = `
<section
  className="sc-fKFyDc gCTAKO"
>
  <button
    className="sc-hBEYos Jeufm"
  >
    <svg
      className="sc-gsTCUz chkPNN"
      fill="#fff"
      fontSize="2.25rem"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 36 19.12"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.87 0L18 16.88 1.13 0 0 1.13 16.88 18 18 19.12 19.13 18 36 1.13 34.87 0z"
        fillRule="evenodd"
      />
    </svg>
  </button>
</section>
`;
